resources/shares/handlers.js

```diff
@@ -0,0 +1,31 @@
+const db = require('../../lib/db')
+const Shares = db.model('Share')
+
+exports.findAll = (req, res) =>
+  Shares
+  .query(qb => qb.limit(req.query.limit).offset(req.query.offset))
+  .orderBy('created_at')
+  .fetchAll({}, { withRelated: ['user'] })
+  .then(collection => res.send(collection))
+
+exports.findOne = (req, res) =>
+  Shares
+  .findById(req.params.id, { withRelated: ['user'] })
+  .then(share => {
+    if (!share) return res.status(404).end()
+    return res.send(share)
+  })
+
+exports.remove = (req, res) => {
+  if (req.params.id !== req.session.user_id) return res.status(403).end()
+
+  return Shares
+  .forge({ id: req.params.id })
+  .destroy()
+  .then(() => res.send({ success: true }))
```

Comment:
Dude, I'm not sure that's the problem, since the sessions handler the following function uncaught: 

```
exports.findOne = (req, res) => {
  User.findById(req.session.user_id)
  .then((user) => res.send(user))
}
```
I've put catches to all my functions and it didn't work